include/jwt-cpp/jwt.h

```diff
@@ -2743,6 +2743,19 @@ namespace jwt {
 		 * \return *this to allow for method chaining
 		 */
 		builder& set_expires_at(const date& d) { return set_payload_claim("exp", basic_claim<json_traits>(d)); }
+		/**
+		 * Set expires at claim to @p d after the issued at claim
+		 * MUST be called after set_issued_at or set_issued_now
```

Comment:
This sounds error prone, I would make this relative to `now()` using a templated clock

Reply:
If the "iat" claim was not set, should this method also set it to the "now" moment?

Reply:
🤔 Regardless of if it is present it should be relative to the `clock.now()`. Having a single behavior would be the best approach

Reply:
Changed this to
```cpp
template<class Rep>
builder& set_expires_in(const std::chrono::duration<Rep>& d) {
	return set_payload_claim("exp", basic_claim<json_traits>(clock.now() + d));
}
```

What type should I use for the duration parameter? Do we consider Clock::now() return such a type that a duration could be added to?

Reply:
Hmmm 🤔 currently it's done with `size_t` and assumes seconds

https://github.com/Thalhammer/jwt-cpp/blob/953aab6fe47be9e92fcbf5fb8418dee0d682e7ca/include/jwt-cpp/jwt.h#L3331

I think this is totally fine, the clock has no concept so I think it's adequate since we already assume substraction https://github.com/Thalhammer/jwt-cpp/blob/953aab6fe47be9e92fcbf5fb8418dee0d682e7ca/include/jwt-cpp/jwt.h#L3154